### PR TITLE
[NO-JIRA] Add Github actions workflow

### PR DESCRIPTION
**Pull request by georgegillams**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-react-native/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: .github/workflows/ios-ci.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: macos-10.15
+
+    steps:
+    # Checkout the repo
+    - uses: actions/checkout@v2
```
Comment (shaundon) [inline .github/workflows/ios-ci.yml]: What does this mean? Like are there docs I can view? It's all good but I'm just trying to wrap my head around how GH Actions works

Comment (georgegillams) [reply .github/workflows/ios-ci.yml]: https://github.com/actions/checkout#checkout-v2

File: .travis.yml [hunk 2]
```diff
@@ -1,107 +0,0 @@
-language: node_js
-
-env:
-  global:
-    - NPM_CONFIG_LOGLEVEL=warn
-    - CLOUDSDK_CORE_DISABLE_PROMPTS=1
-    - ANDROID_API_LEVEL=29
```
Comment (olliecurtis) [inline .travis.yml]: Don't we still need the Android CI right now? Deleting this shouldn't be until we migrate

File: .github/workflows/ios-ci.yml (hunk 1 above)
Comment (shaundon) [reply .github/workflows/ios-ci.yml]: Thanks

File: .travis.yml (hunk 2 above)
Comment (georgegillams) [reply .travis.yml]: Good catch. I deleted this while I was working just to reduce the load on travis. Will reinstate

File: ios/Rakefile [hunk 3]
```diff
@@ -28,9 +28,8 @@ end
 desc "Runs clang-format on all h and m files to ensure formatting is correct"
 task :lint do
   abort red "The following generated files have changed during setup:\n#{get_changed_files}" unless check_pristine
-  `clang-format -i **/*.h **/*.m`
+  `clang-format -i ../**/*.h ../**/*.m`
```
Comment (georgegillams) [inline ios/Rakefile]: To include files in the `lib` directory too

File: .github/workflows/ios-ci.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    - name: brew install
+      run: brew install clang-format
+
+    - name: npm install
+      run: |
+        npm ci
+        if [ -d ./lib/node_modules ]; then cp -rf ./lib/node_modules/* ./node_modules/; fi
```
Comment (tiagoengel) [inline .github/workflows/ios-ci.yml]: Why are we doing this? If npm is putting node_modules there I don't think we should be messing with it. It doesn't seem to cause any problems.

Comment (georgegillams) [reply .github/workflows/ios-ci.yml]: It does, because we assume the location of the font files

Comment (tiagoengel) [reply .github/workflows/ios-ci.yml]: Can you try to rebase? @k0nserv deduped our dependencies to fix that.

Comment (tiagoengel) [reply .github/workflows/ios-ci.yml]: Actually the PR is not merged yet https://github.com/Skyscanner/backpack-react-native/pull/451

Comment (k0nserv) [reply .github/workflows/ios-ci.yml]: Yeah I don't think this should be the solution to this issue. The problem is that we expect the font files to be in `node_modules/bpk-svgs` but they ended up nested under `backpack-react-native` at `node_modules/backpack-react-native/node_modules/bpk-svgs`. I ran `npm dedupe bpk-svgs` in my branch which moves them to the root

File: .github/workflows/ios-ci.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+    - name: bundler install
+      run: |
+        gem install bundler --version 1.17.3
```
Comment (k0nserv) [inline .github/workflows/ios-ci.yml]: This is an old version of bundler, can we upgrade to 2.0?

File: .github/workflows/ios-ci.yml (hunk 4 above)
Comment (k0nserv) [reply .github/workflows/ios-ci.yml]: I just merged #451 so if you rebase on master that should be it

**Comment by k0nserv** [comment]
So I disabled analysis for now since what we are really after here is tests. We need to do more work for analysis because we are getting errors in RN and other libraries atm

**Pull request merged**